src/main/java/core/basesyntax/strategy/impl/OperationHandlerStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationHandlerStrategy;
+import java.util.Map;
+
+public class OperationHandlerStrategyImpl implements OperationHandlerStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> strategies
+            = Map.of(FruitTransaction.Operation.BALANCE, new BalanceOperationHandlerImpl(),
+            FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandlerImpl(),
+            FruitTransaction.Operation.RETURN, new ReturnOperationHandlerImpl(),
+            FruitTransaction.Operation.SUPPLY, new SupplyOperationHandlerImpl());
```

Comment:
also I'd recommend initializing this map in your main method and pass it here using constructor